### PR TITLE
Fix formatting and date issues in file handlers

### PR DESCRIPTION
**Pull request by NicklasHugoy**
* PrevProcessedFile og TempFile benytter den nye nested struktur til smiley reporter
* Et hurtigt (og grimt) fix på at der benyttes to forskellige dato formater nu
* Header i TempFile flushes med det samme for at undgå en tilstand hvor en fil eksitere uden header

### CODE REVIEW

**Comment by NicklasHugoy** [comment]
Har en idé til en anden løsning

File: temp_file.py [hunk 1]
```diff
@@ -58,8 +61,12 @@ def add_data(self, data: dict):
             self.close()
             raise ValueError('Expected data to have "navnelbnr" key')
 
+        dataCopy = data.copy()
+        dump = json.dumps(dataCopy['smiley_reports'])
+        dataCopy['smiley_reports'] = dump
```
Comment (Badgie) [inline temp_file.py]: What is the logic behind this? Seems a bit weird to reassign, but maybe I'm missing something.

**Review by Badgie — APPROVED** [review]
One question, otherwise fine.

File: temp_file.py (hunk 1 above)
Comment (NicklasHugoy) [reply temp_file.py]: To  be able to save the nested dict and list for the smiley reports in the csv file

Comment (NicklasHugoy) [reply temp_file.py]: ah you mean it should be:
`dataCopy['smiley_reports'] = json.dumps(dataCopy['smiley_reports'])`

I was debugging and didn't change it back

Comment (Badgie) [reply temp_file.py]: No no, I was just wondering about the reassignment :D
But ok, that makes sense

**Pull request merged**